token/cli/src/main.rs

```diff
@@ -943,6 +943,21 @@ fn command_burn(
     })
 }
 
+
+fn check_mint(
+    config: &Config,
+    token: Pubkey,
+    ui_amount : f64,
+) -> bool {
+    let supply = config.rpc_client.get_token_supply(&token).unwrap();
+    let amount = ui_amount;
+    if amount + supply.ui_amount.unwrap()> 18446744073709551615.0 {
```

Comment:
When using fixed-point numbers, to easily check for an overflow, you can use `checked_add` and error if that fails: https://doc.rust-lang.org/std/primitive.u64.html#method.checked_add